src/plugins/data/public/query/query_string/language_service/default_language_reference.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+
+import { EuiButtonIcon, EuiLink, EuiPopover, EuiPopoverTitle, EuiText } from '@elastic/eui';
+
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { IDataPluginServices } from '../../../types';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+
+export const DefaultLanguageReference = () => {
```

Comment:
how come we don't register this with the DQL language config:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/query/query_string/language_service/lib/dql_language.ts

Reply:
It did get registered within the DQL language config. Within the language service, when we call createEditor, we pass in the default language reference, and eventually this editor get passed into the DQL language config.

```
 private registerDefaultLanguages() {
    const defaultEditor = createEditor(
      SingleLineInput,
      SingleLineInput,
      [this.createDefaultLanguageReference()],
      DQLBody
    );
    this.registerLanguage(getDQLLanguageConfig(this.defaultSearchInterceptor, defaultEditor));
    this.registerLanguage(getLuceneLanguageConfig(this.defaultSearchInterceptor, defaultEditor));
  }
```